scripts/filter-requirements.py

```diff
@@ -0,0 +1,173 @@
+                    best_tag = select_best_tag(common_tags)
+                    print(f"Best tag selected for {file_info['filename']}: {best_tag}")
+                    return hash_value
+            else:
+                # For source distributions (.tar.gz or .zip),
+                # assume compatibility with all platforms
+                print(f"\nSource distribution {file_info['filename']} is compatible.")
+                return hash_value  # Skip tag matching for source distributions
+
+    return None
+
+
+def extract_tags_from_filename(filename):
+    """Extract tags from a wheel filename, handling multiple platform tags.
+
+    Example filename:
+    aiohttp-3.10.2-cp311-cp311-manylinux_2_5_i686.manylinux1_i686.manylinux_2_17_i686.manylinux2014_i686.whl
+    """
+    # Remove the .whl extension
+    filename = filename.replace(".whl", "")
+
+    # Split the filename into parts, ignoring the first two (package name and version)
+    parts = filename.split("-")
+
+    if len(parts) < 3:
+        print(f"Unexpected filename format: {filename}")
+        return set()
+
+    # Extract Python version and ABI tags (e.g., "cp38-cp38")
+    python_tag = parts[2]
+    abi_tag = parts[3] if len(parts) > 3 else "none"
+
+    # Extract platform tags, which may have multiple parts separated by dots
+    platform_tags = parts[4:]  # Everything after Python and ABI tags
+    platform_tag_list = ".".join(platform_tags).split(
+        "."
+    )  # Split on dots to handle multiple platform tags
+
+    # Create a set of tags by combining the Python tag, ABI tag, and each platform tag
+    tags_set = {
+        f"{python_tag}-{abi_tag}-{platform_tag}" for platform_tag in platform_tag_list
+    }
+    return tags_set
+
+
+def select_best_tag(common_tags):
+    """Generalized function to select the 'best' tag based on versioning and architecture."""
+
+    def tag_priority(tag):
+        """Assign a priority to the tag based on manylinux version."""
+
+        def extract_manylinux_version(tag):
+            """Extract the manylinux version from the tag."""
+            match = re.search(r"manylinux_(\d+)_(\d+)", tag)
+            if match:
+                return int(match.group(1)), int(match.group(2))
+            # Normalize manylinux2014 to manylinux_2_17 and manylinux1 to manylinux_2_5
+            elif "manylinux2014" in tag:
+                return 2, 17
+            elif "manylinux1" in tag:
+                return 2, 5
+            return 0, 0  # Default to lowest version if no match
+
+        # Extract version from the tag
+        manylinux_version = extract_manylinux_version(tag)
+        return manylinux_version
+
+    # Sort the common tags based on the priority function
+    return max(common_tags, key=tag_priority)
+
+
+def get_package_file_info(package_name, package_version, hash_value):
+    """Fetch package file information from PyPI for a specific hash."""
+    url = f"https://pypi.org/pypi/{package_name}/{package_version}/json"
```

Comment:
please make it global constant for now. In future we might "migrate" to Red Hat package index (given it will be created)